cmd/libs/go2idl/deepcopy-gen/generators/deepcopy.go

```diff
@@ -84,38 +137,61 @@ func Packages(context *generator.Context, arguments *args.GeneratorArgs) generat
 		}
 	}
 
-	for _, p := range context.Universe {
-		// Short-circuit if this package has not opted in.
-		if tagvals := types.ExtractCommentTags("+", p.Comments)["k8s:deepcopy-gen"]; tagvals == nil {
-			// No tag for this package.
-			continue
-		} else if tagvals[0] != "generate" && tagvals[0] != "register" {
-			// Unknown tag.
-			glog.Errorf("Uknown tag value '+k8s:deepcopy-gen=%s'", tagvals[0])
+	for i := range inputs {
```

Comment:
I may be missing something, but IIRC, in the previous commit you changed the logic of inputs to only pass directories, for which doc.go file contains the expected tag. That said, if we have a file with only one type opting in for generation (and the whole package not having the tag in doc.go), this will not be processed.
Am I missing something?